control/plugin/metric.go

```diff
@@ -90,8 +90,8 @@ func NewPluginConfigType() ConfigType {
 // Represents a metric type. Only used within plugins and across plugin calls.
 // Converted to core.MetricType before being used within modules.
 type MetricType struct {
-	// Namespace is the identifier for a metric.
-	Namespace_ []core.NamespaceElement `json:"namespace"`
+	// Namespace_ is the identifier for a metric.
+	Namespace_ core.Namespace `json:"namespace"`
```

Comment:
@IRCody, It started from fixing a bug. The change shouldn't affect plugin authors much, but it'll make our code less vulnerable.

Of course, the gain is not much either as plugin authors can still ignore the feature Snap framework tries to provide and use dynamic metric as a static metric unless they specify the name for the dynamic metrics correctly. 

We can document this part. At least, Snap framework supports it.  

If all of you votes against the fix, we can document it as Joel suggested.